From 51e62aaf6941816a08f955bbafdffc6d9281bccb Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 30 Sep 2022 14:49:51 +0300
Subject: [PATCH] [SW-69872] gaudi3: d2d: move d2d ind clear to init

Currently d2d indication clear register is just before the indication
check on slave. Coupling the clear and check together might cause an
issue in case slave executed this logic after master.

Hence as a preventive measure the indication clear is done as a part
of rom init.

Change-Id: I9587acbc07b2abc333c63cd508a723357f6539cd
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 4af10d2..6281d92 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -132,6 +132,10 @@
 	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_5);
 	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_6);
 	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_7);
+
+	/* Die indication register is on die 1 */
+	if (arcboot_rom_global.die_id)
+		regmap_writel(0, mmFFUSE_WRITE_IND);
 }
 
 static void arcboot_rom_set_efuse_read(void)
diff --git a/arch/gaudi3/arc/slave.c b/arch/gaudi3/arc/slave.c
index 6118f15..5e56164 100644
--- a/arch/gaudi3/arc/slave.c
+++ b/arch/gaudi3/arc/slave.c
@@ -66,12 +66,6 @@
 	uint64_t start;
 	uint32_t val;
 
-	if (!arcboot_rom_fuse_d2d_spi_gw_en())
-		return;
-
-	/* Clear indication before polling */
-	regmap_writel(0x0, mmFFUSE_WRITE_IND);
-
 	start = arcboot_rom_get_ts_us();
 	while (1) {
 		val = regmap_readl(mmFFUSE_WRITE_IND);
